### PR TITLE
Gidk 1f7ca3 cf dev| Rohit | custom field implemented for debtor creditors groups accounts

### PR DESCRIPTION
**Pull request by rohit-walkover**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

* **Other information**:

### CODE REVIEW

File: apps/web-giddh/src/app/services/group.service.ts [hunk 1]
```diff
@@ -240,6 +240,39 @@ export class GroupService {
...
+
+     /**
+     * API call to get custom fields for company
+     *
+     * @param {*} model
+     * @returns {Observable<BaseResponse<any, string>>}
+     * @memberof GroupService
+     */
+    public getCompanyCustomField(): Observable<BaseResponse<any, string>> {
+        this.user = this._generalService.user;
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/services/group.service.ts:268]: is this getting used?

File: apps/web-giddh/src/app/services/group.service.ts [hunk 2]
```diff
@@ -240,6 +240,39 @@ export class GroupService {
...
 
+    /**
+     * API call to create custom fields for company
+     *
+     * @param {*} model
+     * @returns {Observable<BaseResponse<any, string>>}
+     * @memberof GroupService
+     */
+    public createCompanyCustomField(model: any): Observable<BaseResponse<any, string>> {
+        this.user = this._generalService.user;
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/services/group.service.ts:251]: is this getting used?

File: apps/web-giddh/src/app/shared/header/components/account-add-new-details/account-add-new-details.component.html [hunk 3]
```diff
@@ -365,6 +365,35 @@ <h3>Address {{i+1}}</h3>
             </div>
 
         </tab>
+
+        <tab *ngIf="isDebtorCreditor" heading="Custom">
+            <ng-container formArrayName="customFields">
+                <ng-container *ngFor="let item of companyCustomFields; let index = index;let f = first; let l = last" [formGroupName]="index">
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/shared/header/components/account-add-new-details/account-add-new-details.component.html:371]: Don't use single character variable

File: apps/web-giddh/src/app/shared/header/components/account-add-new-details/account-add-new-details.component.html [hunk 4]
```diff
@@ -365,6 +365,35 @@ <h3>Address {{i+1}}</h3>
...
+                        <div class="row mb-15" *ngIf="item.uniqueName === customFieldsItem.get('uniqueName').value">
+                            <div class="col-lg-8">
+                                <div class="name-of-field">
+                                    <label>{{item.key}} </label>
+                                    <input *ngIf="item.dataType==='STRING'" type="text" name="values{{index}}" formControlName="value" class="form-control" [maxLength]="item.valueLength">
+                                    <input *ngIf="item.dataType==='NUMERIC'" type="text" name="values{{index}}" formControlName="value" class="form-control" [maxLength]="item.valueLength" decimalDigitsDirective>
+                                    <div *ngIf="item.dataType==='BOOLEAN'" class="clearfix">
+                                        <span class="d-inline-block">
+                                    <input id="yes{{index}}" class="radio-custom cp ml0" type="radio" [checked]="customFieldsItem.get('value').value==='true'"  (change)="selectedBooleanCustomField('true', index)" name="booleanType{{index}}">
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/shared/header/components/account-add-new-details/account-add-new-details.component.html:381]: format the file

File: apps/web-giddh/src/app/shared/header/components/account-add-new-details/account-add-new-details.component.ts [hunk 5]
```diff
@@ -942,4 +946,89 @@ export class AccountAddNewDetailsComponent implements OnInit, OnChanges, AfterVi
...
+
+    /**
+     * To initialize custom field form
+     *
+     * @param {CustomFieldsData} [value=null]
+     * @returns {FormGroup}
+     * @memberof AccountAddNewDetailsComponent
+     */
+    public initialCustomFieldDetailsForm(value: CustomFieldsData = null): FormGroup {
+        let custoFields = this._fb.group({
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/shared/header/components/account-add-new-details/account-add-new-details.component.ts:1001]: correct spelling of custom

File: apps/web-giddh/src/app/shared/header/components/account-update-new-details/account-update-new-details.component.html [hunk 6]
```diff
@@ -354,6 +354,35 @@ <h3>Address {{i+1}}</h3>
             </div>
 
         </tab>
+        <tab *ngIf="isDebtorCreditor" heading="Custom" (select)="tabChanged('custom')">
+            <ng-container formArrayName="customFields">
+                <ng-container *ngFor="let item of companyCustomFields; let index = index;let f = first; let l = last" [formGroupName]="index">
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/shared/header/components/account-update-new-details/account-update-new-details.component.html:359]: Don't use single character variable

File: apps/web-giddh/src/app/shared/header/components/account-update-new-details/account-update-new-details.component.ts [hunk 7]
```diff
@@ -1448,4 +1461,89 @@ export class AccountUpdateNewDetailsComponent implements OnInit, OnDestroy, OnCh
...
+     * @param {*} obj
+     * @param {*} customFieldLength
+     * @memberof AccountUpdateNewDetailsComponent
+     */
+    public renderCustomFieldDetails(obj: any, customFieldLength: any): void {
+        const customField = this.addAccountForm.get('customFields') as FormArray;
+        if (customField.length < customFieldLength) {
+            customField.push(this.initialCustomFieldDetailsForm(obj));
+        }
+        console.log('renderCustomFieldDetails',customField);
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/shared/header/components/account-update-new-details/account-update-new-details.component.ts:1504]: remove console

File: apps/web-giddh/src/app/shared/header/components/account-operations/account-operations.component.html [hunk 8]
```diff
@@ -13,7 +13,7 @@
         <!--share group/account-->
         <div>
             <!--group-->
-            <div class="pull-right col-xs-4 pdR0 text-right">
+            <div class="pull-right col-xs-4 pdR0 pdL0 text-right">
```
Comment (shalinee-walkover) [inline apps/web-giddh/src/app/shared/header/components/account-operations/account-operations.component.html:16]: update pdL0 to pl-0

File: apps/web-giddh/src/app/shared/header/components/account-update-new-details/account-update-new-details.component.html [hunk 9]
```diff
@@ -354,6 +354,35 @@ <h3>Address {{i+1}}</h3>
...
+                                    <input *ngIf="item.dataType==='NUMERIC'" type="text" name="values{{index}}" formControlName="value" class="form-control" [maxLength]="item.valueLength" decimalDigitsDirective>
+                                    <div *ngIf="item.dataType==='BOOLEAN'" class="clearfix">
+                                        <span class="d-inline-block">
+                                    <input id="yes{{index}}" class="radio-custom cp ml0" type="radio" [checked]="customFieldsItem.get('value').value==='true'"  (change)="selectedBooleanCustomField('true', index)" name="booleanType{{index}}">
+                                    <label for="yes{{index}}" class="radio-custom-label">YES </label>
+                                </span>
+
+                                        <span class="d-inline-block">
+                                    <input id="no{{index}}" class="radio-custom cp ml0" type="radio"  [checked]="customFieldsItem.get('value').value==='false'"  (change)="selectedBooleanCustomField('false', index)" name="booleanType{{index}}">
```
Comment (shalinee-walkover) [inline apps/web-giddh/src/app/shared/header/components/account-update-new-details/account-update-new-details.component.html:374]: formate HTML code

File: apps/web-giddh/src/app/shared/header/components/new-manage-groups-accounts/manage-groups-accounts.component.ts [hunk 10]
```diff
@@ -129,4 +175,160 @@ export class ManageGroupsAccountsComponent implements OnInit, OnDestroy, AfterVi
...
+    *
+    * @returns {*}
+    * @memberof ManageGroupsAccountsComponent
+    */
+    public addNewCustomFieldRow(): any {
+        const customRow = this.customFieldForm.get('customField') as FormArray;
+        if (this.customFieldForm.valid) {
+            customRow.push(this.initNewCustomField(null));
+        } else {
+            this.toasterService.warningToast('please fill all mendatory field');
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/shared/header/components/new-manage-groups-accounts/manage-groups-accounts.component.ts:287]: fix spelling of mandatory

File: apps/web-giddh/src/app/shared/header/components/new-manage-groups-accounts/manage-groups-accounts.component.html [hunk 11]
```diff
@@ -26,53 +21,166 @@ <h3 class="modal-title bg pdT">Manage Groups &amp; Accounts</h3>
...
+                                    <label *ngIf="first" class="d-block">&nbsp;</label>
+                                    <button type="submit" class="btn btn-primary" [disabled]="customFieldForm.invalid" [ladda]="isSaveCustomInProgress" (click)="submitCustomFields(customFieldForm.value)">
+                                        Save
+                                    </button>
+                                </div>
+                            </div>
+                        </form>
+                    </ng-container>
+                    <!-- <ng-container *ngIf="!isUpdateModeEnabled">
+                        <form [formGroup]="customFieldForm" (ngSubmit)="submitCustomFields(customFieldForm.value)">
```
Comment (shalinee-walkover) [inline apps/web-giddh/src/app/shared/header/components/new-manage-groups-accounts/manage-groups-accounts.component.html:114]: remove if no need

**Review by ravinderwalkover — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/services/group.service.ts (hunk 1 above)
Comment (rohit-walkover) [reply apps/web-giddh/src/app/services/group.service.ts:268]: yes

Comment (rohit-walkover) [reply apps/web-giddh/src/app/services/group.service.ts:268]: yes in account add and update form

File: apps/web-giddh/src/app/services/group.service.ts (hunk 2 above)
Comment (rohit-walkover) [reply apps/web-giddh/src/app/services/group.service.ts:251]: yes in account add and update form

**Review by shalinee-walkover — APPROVED** [review]
Approve

**Review by ravinderwalkover — APPROVED** [review]

**Pull request merged**